libtransact/src/scheduler/serial/core.rs

```diff
@@ -17,7 +17,7 @@
 
 //! Implementation of core scheduler thread.
 
-use crate::context::manager::ContextManagerError;
+use crate::context::manager::ContextLifecycleError;
```

Comment:
This is a good example of the 'manager' piece of the API leaking into the scheduler implementation. Per suggestion above, this would turn into context::lifecycle::ContextLifecycleError. But if we keep ContextLifecycle in the top-level module, we should just be able to do 'use crate::context::{ContextId, ContextLifecycle, ContextLifecycleError};' here.